### ISSUE TITLE
DDP training hangs when one GPU returns zero loss

### ISSUE THREAD

**Issue by kazimpal87** [PyTorchLightning/pytorch-lightning#12190]
### Discussed in https://github.com/PyTorchLightning/pytorch-lightning/discussions/12189

<div type='discussions-op-text'>

<sup>Originally posted by **kazimpal87** March  2, 2022</sup>
I have a situation where sometimes a batch might contain no ground truth, and so train_step needs to return a zero loss. In the single-gpu case I can just return `torch.tensor([0.0], requires_grad=True, device=self.device)` and it works. However in the DDP case, if one GPU gets a batch like this, it causes the entire training process to hang. I guess this is because this zero tensor doesnt have any gradient information so the reduction doesn't work. Is there some way around this? How can I return zero loss from train_step?</div>

**Comment by akihironitta** [comment]
Duplicate of #12189.
